main.tf

```diff
@@ -20,255 +16,314 @@ data "aws_caller_identity" "current" {
+  cloudtrail_accounts = length(var.cloudtrail_accounts) > 0 ? var.cloudtrail_accounts : [data.aws_caller_identity.current.account_id]
+
+  # if var.cloudtrail_logs_prefix is empty then be sure to remove // in the path
+  cloudtrail_logs_path = var.cloudtrail_logs_prefix == "" ? "AWSLogs" : "${var.cloudtrail_logs_prefix}/AWSLogs"
+
+  # finally, format the full final resources ARN list
+  cloudtrail_resources = toset(formatlist("${local.bucket_arn}/${local.cloudtrail_logs_path}/%s/*", local.cloudtrail_accounts))
+
+  #
+  # Cloudwatch Logs locals
+  #
+
+  # doesn't support logging to multiple accounts
+  # doesn't support logging to mulitple prefixes
+  cloudwatch_effect = var.default_allow || var.allow_cloudwatch ? "Allow" : "Deny"
+
+  # region specific logs service principal
+  cloudwatch_service = "logs.${var.region}.amazonaws.com"
+
+  cloudwatch_resource = "${local.bucket_arn}/${var.cloudwatch_logs_prefix}/*"
+
+  #
+  # Config locals
+  #
+
+  # supports logging to muliple accounts
+  # doesn't support logging to muliple prefixes
+  config_effect = var.default_allow || var.allow_config ? "Allow" : "Deny"
+
+  config_accounts = length(var.config_accounts) > 0 ? var.config_accounts : [data.aws_caller_identity.current.account_id]
+
+  config_logs_path = var.config_logs_prefix == "" ? "AWSLogs" : "${var.config_logs_prefix}/AWSLogs"
+
+  config_resources = sort(formatlist("${local.bucket_arn}/${local.config_logs_path}/%s/Config/*", local.config_accounts))
+
+  #
+  # ELB locals
+  #
+
+  # supports logging to muliple accounts
+  # doesn't support logging to multiple prefixes
+  elb_effect = var.default_allow || var.allow_elb ? "Allow" : "Deny"
+
+  elb_accounts = length(var.elb_accounts) > 0 ? var.elb_accounts : [data.aws_caller_identity.current.account_id]
+
+  elb_logs_path = var.elb_logs_prefix == "" ? "AWSLogs" : "${var.elb_logs_prefix}/AWSLogs"
+
+  elb_resources = sort(formatlist("${local.bucket_arn}/${local.elb_logs_path}/%s/*", local.elb_accounts))
+
+  #
+  # ALB locals
+  #
+
+  # doesn't support logging to multiple accounts
+  # supports logging to multiple prefixes
+  alb_effect = var.default_allow || var.allow_alb ? "Allow" : "Deny"
+
+  # if the list of prefixes contains "", set an append_root_prefix flag
+  alb_include_root_prefix = contains(var.alb_logs_prefixes, "") ? true : false
+
+  # create a list of paths, but remove any prefixes containing "" using compact
+  alb_logs_path_temp = formatlist("%s/AWSLogs", compact(var.alb_logs_prefixes))
+
+  # now append an "AWSLogs" path to the list if alb_include_root_prefix is true
+  alb_logs_path = local.alb_include_root_prefix ? concat(local.alb_logs_path_temp, ["AWSLogs"]) : local.alb_logs_path_temp
+
+  # finally, format the full final resources ARN list
+  alb_resources = sort(formatlist("${local.bucket_arn}/%s/${data.aws_caller_identity.current.account_id}/*", local.alb_logs_path))
```

Comment:
I'm not sure why you have `${data.aws_caller_identity.current.account_id}` as part of the resource arn for things in the bucket with the prefixes/AWSLogs 

Did you mean to have this there?

Reply:
AWS recommends including the account id in the policy ARN

```
      "Resource": "arn:aws:s3:::bucket_name/prefix/AWSLogs/123456789012/*",
```
https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-access-logs.html

We actually do it for the ELB policy, but not on ALB and NLB.

Reply:
ooh okay